jwst/associations/lib/rules_level2_base.py

```diff
@@ -1451,10 +1451,21 @@ def get_opt_element(self):
         """
         item = self.direct_image.item
         opt_elems = []
-        for keys in [["filter", "band"], ["pupil", "grating"]]:
-            opt_elem = getattr_from_list_nofail(item, keys, _EMPTY)[1]
-            if opt_elem:
-                opt_elems.append(opt_elem)
+        inst = str(item.get("instrume", "Unknown"))
+        if inst == "miri":
+            for key in ["filter"]:
+                val = item.get(key)  # we just want to value of 1 key, Use simple get function
+                # getattr_from_list_nofail(item, keys, _EMPTY)[1] is expecting
+                # a pair of keys
+
+                # Check that it exists and isn't the JWST 'null' string
```

Comment:
Under what condition would this not exist or be the NULL string?  And if it were, and you didn't catch it, what would happen?

Reason I ask is that I don't think any of these cases are caught for the other modes, and the asn generator seems to work.  So it's unclear to me as yet why you need to worry about these catches for MIRI.

Maybe it's more a @tapastro question as to what happens if these optical elements don't match what's expected for a given instrument.